Update S3 config to get media and static files working

amygori opened the pull request:
This moves us in the direction of using the Momentum S3 bucket for static and media files. Notable change here include:
- new `STORAGES` setting that replaces the old `DEFAULT_FILE_STORAGE` setting in Django versions prior to 4.2.
- new storage backend classes
- set the default ACL on AWS to `public-read`

**Caveats**
This needs to be tested against the front end before it's ready to deploy. No tests are included in this PR. 😬

amygori committed:
Change storages config for Django >= 4.2 compatibility

This commit updates the storages setting to use the new default backend.

It also sets the default public ACL on S3 to true to allow for public access of images.

More info about configuring django-storages:
https://django-storages.readthedocs.io/en/latest/backends/amazon-S3.html#configuration-settings

About the STORAGES setting in Django:
https://docs.djangoproject.com/en/4.2/ref/settings/#storages

amygori committed:
Add UserAdmin class for better user password management in admin

The big benefit is that passwords are properly hashed when users are added via the admin.
More about customizing the UserAdmin class here:
https://docs.djangoproject.com/en/4.2/topics/auth/customizing/#custom-users-and-django-contrib-admin

amygori committed:
Merge branch 'staging' into s3-storage-config

GitLukeW commented on config/settings.py:
```diff
@@ -156,23 +156,30 @@
+    AWS_DEFAULT_ACL = 'public-read'
+    AWS_QUERYSTRING_AUTH = False
+    AWS_S3_FILE_OVERWRITE = False
+    STORAGES = {
```
The custom user model is setup to pull from django.core.files.storage import default_storage. With this new change I think we will need to update the import so it's pointing to storages now. 

To recreate the error you can run `python manage.py test`. This should fail any test looking for default storage.

amygori replied:
Ah nice! I will run the tests and check it out, and update the PR accordingly. Thanks for the heads up @GitLukeW!

GitLukeW reviewed (changes requested):
Thank you for the contribution!!! Added one request to change an import name now that the storages name has changed.

GitLukeW committed:
Merge branch 'staging' into s3-storage-config

GitLukeW committed:
Merge branch 'staging' into s3-storage-config

amygori commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Use new AWS account in production for custom and default profile photos](https://trello.com/c/wdnTY5bJ/158-use-new-aws-account-in-production-for-custom-and-default-profile-photos)

amygori committed:
Use Django's file storage API instead of boto3 so this will work in dev as well

amygori committed:
Add tests for user profile photo existence and storage

amygori committed:
Merge remote-tracking branch 'upstream/staging' into s3-storage-config

amygori commented:
Updated this PR to include tests 🎉! Changed settings in AWS aren't reflected in this PR but I think there was just a missing Read permission on existing objects in the bucket. We may want to update the way permissions are set, however, to be in line with [current AWS guidance](https://docs.aws.amazon.com/AmazonS3/latest/userguide/security-best-practices.html) to use a policy rather than ACLs. 

I manually tested against a local front-end instance and encountered a few bugs on the FE but  the requests appear to be working with the BE updated.

GitLukeW reviewed (approved):
Looks good, can't wait to get it into staging! Thank you for the contributing to the project!
